UI/FleetWnd.cpp

```diff
@@ -1534,7 +1534,9 @@ void FleetDataPanel::Refresh() {
         {
             add_overlay("bombarding.png");
         }
-        if (fleet->OrderedGivenToEmpire() != ALL_EMPIRES)
+
+        // Moving fleets can't be gifted.
+        if (fleet->OrderedGivenToEmpire() != ALL_EMPIRES && fleet->TravelRoute().empty())
```

Comment:
this doesn't make sense... if the situation somehow arises that a moving fleet is being gifted, it should be displayed. enforcing such a rule should be done where the gifting order is issued and where it is executed, not just when displaying the indicator.

Reply:
The rule is enforced on the server in the gifting code.

However, in the UI without this line of code, gifting a ship and then moving it still shows the gifting icon, even though the gift will not happen on the server.  This code makes the result of the order set be reflected in the UI.

 I added a comment to clarify this.

Another approach is the cancel the gifting when creating the move order in the UI, but then the order needs to be manually restored when a fleet is stopped.  I found that the proposed  approach takes fewer clicks to cancel a gifting order by moving a fleet and then restore the order by stopping the fleet from moving.  In addition, it is clear that the moving is cancelling the order because stopping the fleet restores the gifting icon.